### DISCUSSION TITLE
esp32_cam change streaming port 81 to 80

### DISCUSSION THREAD

**Discussion by tgreif** [arendst/Tasmota#12523] [Q&A] [answered]
hi everybody . 
i have flashed the esp32_cam to my board .
its working ..HURA 
but because i 'm using a nrok tunnel , i need to change the video streaming port from 81 to 80 . 
somebody knows if it is possible ? 
i've tryed to change it in "xdrv_81_esp32_webcam.ino" but without succed .

Tasmota 9.3.1.1

**Comment by sfromis** [comment]
I think that your problem is that Tasmota already has a web server on port 80, where dual use would take integrating the code for the two web servers, which is probably not straightforward. One thing to try would be running Tasmota without the regular web server on port 80, which may allow the streaming server to operate on port 80. One Tasmota config line is:
`#define WEB_PORT             80                // Web server Port for User and Admin mode`

Whether it will work may be another question, as a tool *requiring* the use of port 80 may be designed for http only.

**Reply by tgreif** [reply]
thank you , 
the problem is that on the esp32_cam module , the stream is only aviable using the "http://my ip:81" .  so on a node-red dashbord in local application it's ok. but if yiou're outlite the local , using a ngrok  with limitation on port 80 , the dashboard is not able to show the stream ... 
i'm open for any solution ...

**Reply by sfromis** [reply]
You tried changing the driver to port 80, and it did not work. Did you try it without having the regular Tasmota web server on port 80?

**Reply by barbudor** [reply]
ngrok is not limited to port 80
You can create a tunnel on port 81

**Reply by tgreif** [reply]
> You tried changing the driver to port 80, and it did not work. Did you try it without having the regular Tasmota web server on port 80?

no i've not tryed yet i will try this afternoon.

**Reply by tgreif** [reply]
> ngrok is not limited to port 80
> You can create a tunnel on port 81

i will try to open the tunnel on port 81 ., and see if i can see the stream in the node-red dashboard. 
thanks

**Comment by tgreif** [comment, marked as answer]
hi Barbudor , 
I've tryed to change to port 81 , 
yes it's possible but I 've only a free account so it is limited to only one port . 
I will dicribe the situation better . 
my raspberry runs node_red an mosquitto  on a local network . for to access the network from outside i've the ngrok tunnel . in the local network are different es8266 like shutter and swiches and sensor, and one esp32 _cam for the video stream . the video stream is good from the local but not good from outside . (node red Dashboard)

 ngrok is configured for the :80  so the dashboard is working . 
i'm not able to open a second tunnel 0n port 81 !??


perhaps i do a different mistake ? 

is there any other way to show the video stream from outside the network?

**Reply by sfromis** [reply]
My starting point would be to create a private VPN server

**Comment by tgreif** [comment]
hi everybody . 

iv tryed to change the port for the videostremo in tasmota firmware to 80 and changed the webserver port to 90 . 
until here it works. . 
in the node-red i have changed to "img src="http://192.168.1.6:80/stream" width="500px""
and on local network it work. from outside the Lan it do not work . so it is not a problem of the port but....??
any ideas 
vpn is posisbile but i hpe to find the way to make it working without!

**Reply by ascillato** [reply]
From an external access is not a Tasmota issue anymore. So, now it is a networking challenge. How are you trying to forward the port? If you are opening the port 80 on your router you will have 3 problems at least:
+ Most ISP blocks port 80 for security
+ Most routers don't allow to use 80 due to is used for them for routers configuration.
+ 80 is one of the most attractive port for hackers trying to take control of your devices.

So. Port forwarding is never a good idea without very good knowledge of what are you doing.

The best option is use a hub that gather all your information, camera feeds, devices, etc like home assistant and secure just one instance.

**Comment by tgreif** [comment]
Hi ascillato , yes you 're right . its a network challenge . On the tasmota side everything is working well . 
I will try to describe my infrastructure . first of all, it is in my house in the countryside. there is no phone or ADSL but only a mobile acesspoint from d-link with 4G. so no possibility for a static IP . 
on my router i ve coneccted a raspberry 4, running mosquitto and None-red. on ip 192.168.1.202. also the ngrock tunnel on port 80. 
the raspy is accessible from outside the local network. 
ive different self made boards , based on esp822 (see the foto) , and one esp32_cam.  
on my dashboard you can see staus , switches anc the sutter. from the local as from outside .the video stream form the esp32_cam is only visible from the lan . from outside not . there is no portforwading or other on my router because i use teh ngrok tunnel .
attached you find the instruction for the node_red viedeo stream 
 my question is : how i can make visible the video stream from outside the lan ??


![Cattura3](https://user-images.githubusercontent.com/2463499/124277364-40b8a080-db45-11eb-97fe-d342f7f7c36e.JPG)

![20210702_144236](https://user-images.githubusercontent.com/2463499/124276603-4feb1e80-db44-11eb-9942-4763f680e63e.jpg)

**Reply by barbudor** [reply]
You need a ngrok tunnel to port 81 of the esp32cam and this  one you should refer to in the NodeRed sashboard, not your local address

**Comment by tgreif** [comment]
ok yes that could be a soliution but at this point it wil be possible to use the raspberry pi also as server for the viedeostream?? 
the raspberry has its ngrok tunnel . 
do you thik it will be pssible ?

**Reply by Jason2866** [reply]
What is running on the Rpi4? Install a Home Automation system on it and do what ascillato already suggested. One central point where everything is routed through. You have way more possibilitys to solve what you want to do.
You could use a DynDNS service. So connecting from outside via a VPN tunnel is no problem anymore. You have no restrictions doing network stuff.

**Reply by ascillato** [reply]
My own setup doesn't use any port forwarding or anything like that. Maybe you can simplify your setup.

I use an ESP32CAM with Tasmota (standard version without modifications)
On a Raspberry Pi, I run Home Assistant. And in HA I use the official HA Add-On called MotionEye. I have configured MotionEye to save video streams both locally and in the cloud.
Then I simply connect to Home Assistant from my phone outside of my network using Home Assistant's Nabu Casa connection and I can seamlessly view my ESP32CAMs.

As most of the open source home automation software are easily scalable, you can add many more features like person recognition, notifications, etc.

**Reply by defaliz** [reply]
that's work for me with raspberry and autossh, you can mount tunnel ssh 
or you can use nginx proxypass, boths works for me...

**Comment by tgreif** [comment]
Ok that semestre to be a solutions,  but i'm usino node red . Is there a simile solutions?

**Reply by defaliz** [reply]
i don't know tgreif, i don't use node red..
